dotnet/src/Microsoft.Agents.AI.Workflows.Declarative/CodeGen/AddConversationMessageTemplate.cs

```diff
@@ -76,23 +77,28 @@ public override string TransformText()
             AgentMessageContentType contentType = content.Type.Value;
             if (contentType == AgentMessageContentType.ImageUrl)
             {
-            this.Write("\n        yield return new UriContent(contentValue, \"image/*\");");
+            this.Write("\n        content.Add(UriContent(contentValue");
+            this.Write(this.ToStringHelper.ToStringWithCulture(index));
+            this.Write(", \"image/*\"));");
 
             }
             else if (contentType == AgentMessageContentType.ImageFile)
             {
-            this.Write("\n        yield return new HostedFileContent(contentValue);");
+            this.Write("\n        content.Add(new HostedFileContent(contentValue");
```

Comment:
What kind of data is ImageFile? HostedFileContent represents a server-side file that the client just references by an ID. If ImageFile is instead bytes that are passed around (maybe as a data uri or as base64-encoded content), it should be using DataContent.

Reply:
> For data encoded url, I probably need to update the logic in the UriContent handler (and I think multiple locations).
> 
> #1086

Yeah. You _can_ put a data URI into a UriContent... the problem is that, before .NET 10, System.Uri has a length limitation, so once you get beyond ~50K of data being encoded, it'll blow up. DataContent can be constructed from a data URI and it'll "just work".